docs/resources/task.md

```diff
@@ -0,0 +1,224 @@
+# Task Resource
+
+This resource will:
+
+1. Create cloud resources (storage, machines) for the task
+2. Upload the given `directory` to the cloud storage, if specified
+3. Run the given `script` upon completion or `timeout` in the cloud
+
+## Example Usage
+
+```hcl
+resource "iterative_task" "task" {
+  name  = "example"
+  cloud = "aws"
+
+  environment = {GREETING = "Hello, world!"}
+  directory   = "${path.root}/shared"
+
+  script = <<-END
+    #!/bin/bash
+    echo "$GREETING" | tee $(uuidgen)
+  END
+}
+```
+
+## Argument Reference
+
+### Required
+
+- `name` - (Required) Task name.
+- `cloud` - (Required) Cloud provider to run the task on; valid values are `aws`, `gcp`, `az` and `k8s`.
+- `script` - (Required) Script to run; must begin with a valid [shebang](<https://en.wikipedia.org/wiki/Shebang_(Unix)>).
```

Comment:
Users can always resort to [`file`](https://www.terraform.io/docs/language/functions/file.html) to use a file instead of an inline script. Would you like to include this in the argument description? It looks like a nice addition.

Reply:
```suggestion
- `script` - (Required) Script to run; must begin with a valid [shebang](<https://en.wikipedia.org/wiki/Shebang_(Unix)>); can be specified as a [heredoc](https://www.terraform.io/docs/language/expressions/strings.html#heredoc-strings) or loaded from a separate [`file`](https://www.terraform.io/docs/language/functions/file.html).
```

Reply:
74cedc0